### ISSUE TITLE
Possible regression in 0.9.36

### ISSUE THREAD

**Issue by davidwessman** [glebm/i18n-tasks#408]
Hello!

I have a translation that is caught by `i18n-tasks missing` incorrectly in 0.9.36 but not in 0.9.35.

I reproduced it in the gem's test suite by adding:

```slim
section.row-fluid
  .span12
    - @form.section.each do |section|
      - if section.queston.select { |obj| (obj["type"] == "A" || obj["type"] == "B" || obj["type"] == "C") }.length > 0
        = render("fields",
                 section: section,
                 form: @form)
```

to the end of `spec/fixtures/app/views/index.html.slim`.

When it tries to match

```slim
= t "a.b", scope: "scope.subscope"
```

it will not get `a.b` but the rest of the file, but only if I add my code afterwards.

It passes on 0.9.35 but not on 0.9.36.

**Comment by davidwessman** [comment]
Based on a `git bisect` this was introduced in [0885805912fbd7450ac4ccc74e1bac6e11b65d1a](https://github.com/glebm/i18n-tasks/commit/0885805912fbd7450ac4ccc74e1bac6e11b65d1a)

@otoyo Do you have any guesses to what we need to change?

**Comment by otoyo** [comment]
@davidwessman I have pushed https://github.com/glebm/i18n-tasks/pull/410 to revert.

**Comment by glebm** [comment]
@otoyo For more robust parsing of ERB, it may be better to write a separate `ErbScanner` class. The scanner could run an actual Ruby parser on the `<% %>` blocks. With some refactoring it may be possible to reuse most of `RubyAstScanner` for that.

**Comment by otoyo** [comment]
Yes, I think so too. I should have been separate regexp to `ErbScanner`.
Thank you for merging fix.

**Comment by davidwessman** [comment]
How can we make it work for slim as well as erb? The code I sent triggered this in a slim file. Please reach out if you want my help to test anything. Thank you for your work!

**Comment by glebm** [comment]
Released v0.9.37 for now with the revert
